src/GraphQL/GraphQLBuilderExtensions.cs

```diff
@@ -0,0 +1,827 @@
+        /// registers clr type mappings on the schema between that class and the source type or underlying enum type.
+        /// Skips classes where the source type is <see cref="object"/>, or where the class is marked with
+        /// the <see cref="DoNotMapClrTypeAttribute"/>.
+        /// </summary>
+        /// <remarks>
+        /// This is equivalent to calling <see cref="SchemaExtensions.RegisterTypeMappings(ISchema)"/>
+        /// within the schema constructor.
+        /// </remarks>
+        public static IGraphQLBuilder AddClrTypeMappings(this IGraphQLBuilder builder)
+            => builder.AddClrTypeMappings(Assembly.GetCallingAssembly());
+
+        /// <summary>
+        /// Scans the specified assembly for classes that inherit from <see cref="ObjectGraphType{TSourceType}"/>,
+        /// <see cref="InputObjectGraphType{TSourceType}"/>, or <see cref="EnumerationGraphType{TEnum}"/>, and
+        /// registers clr type mappings on the schema between that class and the source type or underlying enum type.
+        /// Skips classes where the source type is <see cref="object"/>, or where the class is marked with
+        /// the <see cref="DoNotMapClrTypeAttribute"/>.
+        /// </summary>
+        /// <remarks>
+        /// This is equivalent to calling <see cref="SchemaExtensions.RegisterTypeMappings(ISchema, Assembly)"/>
+        /// within the schema constructor.
+        /// </remarks>
+        public static IGraphQLBuilder AddClrTypeMappings(this IGraphQLBuilder builder, Assembly assembly)
+        {
+            if (assembly == null)
+                throw new ArgumentNullException(nameof(assembly));
+
+            // retreive all of the type mappings ahead-of-time, in case of a scoped or transient schema,
+            // as reflection is relatively slow
+            var typeMappings = assembly.GetClrTypeMappings();
+            builder.ConfigureSchema((schema, serviceProvider) =>
+            {
+                foreach (var typeMapping in typeMappings)
+                {
+                    schema.RegisterTypeMapping(typeMapping.ClrType, typeMapping.GraphType);
+                }
+            });
+
+            return builder;
+        }
+        #endregion
+
+        #region - AddDocumentListener -
+        /// <summary>
+        /// Registers <typeparamref name="TDocumentListener"/> with the dependency injection framework as both <typeparamref name="TDocumentListener"/> and
+        /// <see cref="IDocumentExecutionListener"/>. Configures document execution to add an instance of <typeparamref name="TDocumentListener"/> to the
+        /// list of document execution listeners within <see cref="ExecutionOptions.Listeners"/>. Singleton, scoped and transient lifetimes are supported.
+        /// </summary>
+        /// <remarks>
+        /// Do not separately add the document listener to your execution code or the document listener may be registered twice for the same execution.
```

Comment:
:) I think you already expected danger from double registrations.

Reply:
It will be the case for AddMiddleware and all code that uses ConfigureSchema or ConfigureOptions.  (e.g. AddMetrics).  I didn't add notes everywhere, but I did so here.  I really think that the builder is for new code more than for old code.  Or else users need to understand the amount of configuration that the builder methods perform automatically, and hence what they need to remove from their old code if they intend to use the builders.